### ISSUE TITLE
Save state slot number is the same for all games

### ISSUE THREAD

**Issue by geniot** [plrguez/fuse-od#3]
Probably not a bug. But still worth mentioning. If in one game you proceed to save 15 states and then you quit to play another game and then you start saving states the suggested slot number is 15 (last one used in the previous game), not 1,2,3 or the first available one as expected. 
If it is expected of course :)

**Comment by plrguez** [comment]
It's not a bug, but it's a good suggestion and I will take it in account.

This could be converted in a new option for save states: Not reset slot number on program change (current behaviour), reset to to the first slot or reset to the first free slot (or the first if all slots have been used).

**Comment by geniot** [comment]
I think this option would make sense if there were 100 slots total for all games. 
Anyway it's a matter of taste. Eg. what slot to select by default when entering 'Load state'? The first one or the last one? How to display the list: starting from the first item or the last one at the bottom? Same for save. And what if there are empty slots in between? Too many questions. 

One thing that wasn't immediately evident to me when I started to use Fuse on PocketGo was that 'A' confirms the changes in options like checkboxes checked and 'B' actually cancels them. So everytime I used 'B' to return, because I thought that 'A' selects something and 'B' is just return (with selections enabled). There was no feel of that modal dialog with Ok and Cancel. But then after minutes of testing I learned how things work.

**Comment by plrguez** [comment]
Yes, probably too many questions. Well, when I get to work on it, I'll think about it more thoroughly.

In the the Fuse's SDL UI Enter key is to select options and confirm form dialogs, Escape key to cancel and Space key to mark check options or open listbox options.
The form dialogs must be confirmed, if you cancel it with Escape all the changes made in form are lost.
The openDingux port use the same SDL UI, and the 'A' button is like Enter, 'B' button is like Escape and 'X' button like Space.
